specification/consumption/resource-manager/Microsoft.Consumption/stable/2023-03-01/consumption_pricesheet.json

```diff
@@ -0,0 +1,164 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-03-01",
+    "title": "ConsumptionManagementClient",
+    "description": "Consumption management client provides access to consumption resources for Azure Enterprise Subscriptions."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingPeriods/{billingPeriodName}/providers/Microsoft.Consumption/pricesheets/download": {
```

Comment:
>/providers/Microsoft.Consumption/pricesheets/download [](http://example.com/codeflow?start=103&length=53)

The resource path is not valid.

It should either be:

POST "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingPeriods/{billingPeriodName}/providers/Microsoft.Consumption/pricesheets/{pricesheetName}/download"

or

POST "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/billingPeriods/{billingPeriodName}/providers/Microsoft.Consumption/downloadPricesheets"

Reply:
agreed, unfortunately, since this is an existing API, it cannot be modified at this point.

Reply:
Is this implemented recently and not exposed to customers yet? If yes, please update the API and its implementation. This API path doesn't comply with ARM RPC, and needs to be updated.

Reply:
No, this has been around for a long time now and is being used by customers. This documentation was missed somehow. Hence the limitation.